R/ppc-distributions.R

```diff
@@ -62,6 +97,57 @@
 #'    See Säilynoja et al. (2021) for more details.}
 #' }
 #'
+#' @section When to Use Each Plot:
+#'
+#' These plots offer different perspectives on how simulated (`yrep`) and observed (`y`) data compare.
+#'
+#' \itemize{
+#'   \item **`ppc_hist()`** — Best for visualizing **overall distribution shapes**.  
+#'   Use this when your data are continuous and you want to see how model predictions align
+#'   with observed distributions.
+#'
+#'   \item **`ppc_dots()`** — Ideal for **small or discrete datasets**.  
+#'   Displays each simulated dataset as dots, giving an intuitive visual comparison
+#'   between simulated and observed values.
+#'
+#'   \item **`ppc_freqpoly()`** — Useful for **moderate to large continuous datasets**.  
+#'   Highlights how predicted and observed **frequencies vary** across the range of `y`.
+#'
+#'   \item **`ppc_freqpoly_grouped()`** — When you have a **grouping variable** (e.g., categories),
+#'   use this to compare frequencies across levels of that variable.
+#' }
+#'
+#' @section Usage Note:
+#'
+#' - The `binwidth`, `bins`, and `breaks` arguments are passed to multiple
+#'   underlying `ggplot2` and `ggdist` geoms (`geom_histogram()`, `geom_area()`,
+#'   and `stat_dots()`), which means they affect both histogram and dot plot outputs.
+#'
+#' - The `freq` argument controls whether the y-axis represents **counts**
+#'   (`freq = TRUE`) or **densities** (`freq = FALSE`) for both histograms
+#'   and frequency polygons.
+#'
+#' - These plots assume numeric `y` values. For categorical data,
+#'   consider `ppc_bars()` or similar discrete comparison functions.
+#'
+#' - If a grouping variable is provided, ensure that `y` and `yrep` have
+#'   matching group structures; otherwise, plots may not align correctly.
+#'
+#' - For large replicated datasets (`yrep` with > 50 rows), consider
+#'   summarizing or subsampling to avoid overplotting.
+#'
+#' Note:
+#' The `binwidth`, `bins`, and `breaks` arguments are forwarded not only
+#' to `ggplot2::geom_histogram()` (used by `ppc_hist()`), but also to
+#' `ggdist::stat_dots()` (used by `ppc_dots()`) and to the `stat = "bin"`
+#' layers used by `ppc_freqpoly()` and `ppc_freqpoly_grouped()`
+#' (implemented with `ggplot2::geom_area(stat = "bin", ...)`).
+#'
+#' Similarly, the `freq` argument affects not only histograms but also
+#' frequency polygons created with `ppc_freqpoly()` and
+#' `ppc_freqpoly_grouped()`, where it determines whether the y-axis
+#' represents counts (`freq = TRUE`) or densities (`freq = FALSE`).
```

Comment:
This is all good info but I think most of it is clear from the existing documentation and it adds a lot of length to already long documentation. For example, it should already be clear that the `freq` argument affects multiple different plots because they each have the `freq` argument. I'm not sure we need to write that out in  sentence form. 

I guess my main concern here is that users are just going get lost in a sea of text. So we should only put things in sentences/paragraphs that are hard to discern from the existing documentation. Does that make sense? If you disagree feel free to let me know! I really appreciate the effort you put into this.